packages/http/src/payload.ts

```diff
@@ -501,26 +526,18 @@ function resolvePart(
   return diagnostics.wrap(undefined);
 }
 
-function getFilePart(
-  name: string | undefined,
-  file: HttpFileModel,
-): DiagnosticResult<HttpOperationPart | undefined> {
+function getFileBody(file: HttpFileModel): DiagnosticResult<HttpOperationFileBody> {
   const [contentTypes, diagnostics] = getContentTypes(file.contentType);
+
   return [
     {
-      multi: false,
-      name,
-      body: {
-        bodyKind: "single",
-        contentTypeProperty: file.contentType,
-        contentTypes: contentTypes,
-        type: file.contents.type,
-        isExplicit: false,
-        containsMetadataAnnotations: false,
-      },
+      bodyKind: "file",
+      type: file.type,
+      contents: file.contents,
       filename: file.filename,
-      optional: false,
-      headers: [],
+      isText: file.contents.type.name === "string",
```

Comment:
Should we allow scalars that might extend string to be treated as text?

Reply:
good call, yes. this is definitely an oversight.

Reply:
Made this much more robust